src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -411,6 +411,33 @@ private BoundExpression BindToInferredDelegateType(BoundExpression expr, Binding
                 }
                 delegateType = CreateErrorType();
             }
+            else
+            {
+                // If synthesizing a delegate with `decimal`/`DateTime` default value,
+                // check that the corresponding `*ConstantAttribute` is available.
```

Comment:
>check that the corresponding `*ConstantAttribute` is available. [](http://example.com/codeflow?start=19&length=63)

We don't need to check for ParamArray attribute on this code path?

Reply:
It's already checked in `SourceParameterSymbol.Create` for lambda/method definitions and `Conversions.GetMethodGroupDelegateConversion` for conversions to synthesized delegates. Originally, I wanted to put checking for `*ConstantAttribute`s there, as well, but there's not enough information in those places (like default value). But I guess I could move `ParamArray` checks here instead.